### PR TITLE
Disable Local Push Notifications + Start Handling invite notifications

### PR DESCRIPTION
**Pull request by Velin92**
Local Push notifications are quite a niche case, so we don't really need that much, also sometimes can be a bit flaky, so we decided to disable them on iOS.

Also
- did some refactor in the code for push notifications
- started implementing the `processInvite` function for invite notifications (however they can't be properly handled yet since the NotificationItem that generated them throws an error)
- Finally found the fix for the missing avatars
- Removed the reply action  in the notification since we don't support it yet

### CODE REVIEW

File: ElementX/Sources/Other/Extensions/UNNotificationContent.swift [hunk 1]
```diff
@@ -127,7 +127,7 @@ extension UNMutableNotificationContent {
         if let mediaSource = iconType.mediaSource {
             switch await mediaProvider?.loadFileFromSource(mediaSource) {
             case .success(let mediaFile):
-                image = try INImage(imageData: Data(contentsOf: mediaFile.url))
+                image = try await INImage(imageData: Data(from: mediaFile.url))
```
Comment (stefanceriu) [inline ElementX/Sources/Other/Extensions/UNNotificationContent.swift:130]: Shouldn't you use the `MediaLoaders` `loadMediaContentForSource` here instead?

File: ElementX/Sources/Services/Client/ClientProxy.swift [hunk 2]
```diff
@@ -540,7 +544,7 @@ class ClientProxy: ClientProxyProtocol {
         if let notificationsListBuilder {
             MXLog.info("Registering notifications view")
             _ = slidingSync?.addList(listBuilder: notificationsListBuilder)
-        } else {
+        } else if ServiceLocator.shared.settings.enableLocalPushNotifications {
```
Comment (stefanceriu) [inline ElementX/Sources/Services/Client/ClientProxy.swift:547]: Should you maybe wrap the whole `if let notificationsListBuilder...` block in this check instead?

**Review by stefanceriu — APPROVED** [review]
Left some comments inline, otherwise lgtm 👍

File: ElementX/Sources/Services/Client/ClientProxy.swift (hunk 2 above)
Comment (Velin92) [reply ElementX/Sources/Services/Client/ClientProxy.swift:547]: oh yeah that's better

File: ElementX/Sources/Other/Extensions/UNNotificationContent.swift (hunk 1 above)
Comment (Velin92) [reply ElementX/Sources/Other/Extensions/UNNotificationContent.swift:130]: I have the MediaProviderProtocol here

Comment (Velin92) [reply ElementX/Sources/Other/Extensions/UNNotificationContent.swift:130]: oh wait you suggest actually instantiating a MediaLoader and using it in place of creating the data? Ok let's give it a try honestly I am not a fan of this AI generated solution of using a URLSessionProtocol

**Pull request merged**